### PR TITLE
Text array

### PR DESCRIPTION
**Pull request by g--o**
Adds the text array with 3 quick test levels
Scriptable & binded to the menu selection and text deletion (defaults: ENTER & BACKSPACE)

### CODE REVIEW

**Comment by g--o** [comment]
dammit forgot to change name in header... I'll fix it soon enough

File: src/object/text_array_object.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+void TextArrayObject::set_fade_transition(bool fade_transition)
+{
+  m_fadeTransition = fade_transition;
+}
+
+TextArrayItem* TextArrayObject::get_text_item(ta_index index)
+{
+  if (m_texts.size() <= 0)
+    return nullptr;
+  return this->m_texts.at(index).get();
```
Comment (christ2go) [inline src/object/text_array_object.cpp]: What happens, if index is out of bounds?

File: src/object/text_array_object.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+typedef unsigned int ta_index;
+
+/** A text object intended for scripts that want to tell a story */
+class TextArrayObject : public GameObject,
+                        public ExposedObject<TextArrayObject, scripting::TextArray>
+{
+public:
+  TextArrayObject(const std::string& name_ = std::string());
+  ~TextArrayObject() = default;
+
```
Comment (christ2go) [inline src/object/text_array_object.hpp]: It would be great to have deoxygenated comments explaining what the methods do.

**Comment by tobbi** [comment]
Hey, I'm wondering: Do these .sh files need to be there? Normally we manage all the build stuff using CMake.

File: ubuntu-dev-installs.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (christ2go) [inline ubuntu-dev-installs.sh]: Please don’t add unrelated files.

**Comment by g--o** [comment]
@tobbi It's for convenience. If we don't want it I can remove them.

**Comment by tobbi** [comment]
@g--o What do they do?

**Comment by g--o** [comment]
@tobbi They're just for setting up the environment for ubuntu quickly.
On your first time you run ubuntu-dev-installs.sh to install all the lib packages.
And then you run build.sh to build (creates dir, runs cmake + make, adds flags)
As said, it's not neccessary, but I find it comfy and that's how I work anyway.

**Comment by g--o** [comment]
@christ2go Fixed.
About the editor integration - I'm not sure I understand, do you mean an invisible block or a some dialog for inserting one? If so I'd prefer to do this in a separate PR, along with the picnic scene. For now it's scriptable like the text object. You can check the test levels to see some examples.

File: src/object/text_array_object.cpp [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+{
+  if (m_isDone)
+    return;
+
+  // make sure there's anything to update
+  if (m_texts.size() <= 0)
+    return;
+
+  // detect change request
+  InputManager* current = InputManager::current();
```
Comment (tobbi) [inline src/object/text_array_object.cpp]: Can you use ```auto current =```

**Comment by g--o** [comment]
@tobbi should be good now

**Comment by tobbi** [comment]
It looks good, I just have a few questions:
* The white background looks rather opaque on the first item of the text array. Any ideas?
* Could we have it so that you can specify a duration for each text item (maybe in addition to specifying a default duration).

**Comment by g--o** [comment]
@tobbi 
- Could be because of differences in fade_in vs. visible(true), I'm not sure and I honestly just forward the API to the relevant item.
- But you do have that, it's even in one of the tests =/

**Comment by tobbi** [comment]
I'll give this one final test and if all is good, I'll merge it.
